### PR TITLE
Fix PCR tests and added tests from CMM

### PR DESCRIPTION
**Pull request by Fletch153**

### CODE REVIEW

**Review by dsshap — CHANGES_REQUESTED** [review]
@Fletch153 looks like not all the tests transcribed correctly. 

Lets move the lower strike tests to a separate contract so that we dont need to always setSpot on the higher strike tests (which there a lot more of). 

Also, please take a look through all the tests to make sure the balances match what is in CrossMarginMath.t.sol

File: src/test/engine-integrations/cross-margin/PreviewCollateralReq.t.sol [hunk 1]
```diff
@@ -16,219 +16,446 @@ import "../../../core/engines/cross-margin//AccountUtil.sol";
...
-    //     shorts[3] = Position(c26000, uint64(8 * UNIT));
-    //     shorts[4] = Position(c27000, uint64(1 * UNIT));
+        Balance[] memory balances = _previewMinCollateral(positions);
 
-    //     longs = new Position[](2);
-    //     longs[0] = Position(p17000, uint64(1 * UNIT));
-    //     longs[1] = Position(c25000, uint64(16 * UNIT));
+        assertEq(balances.length, 1);
+        assertEq(balances[0].collateralId, wethId);
+        assertEq(balances[0].amount, ((100000 * 1000 * UNIT) / 22000) * 1e12);
```
Comment (dsshap) [inline src/test/engine-integrations/cross-margin/PreviewCollateralReq.t.sol:236]: @Fletch153 what currency is `1e12`?

File: src/test/engine-integrations/cross-margin/PreviewCollateralReq.t.sol [hunk 2]
```diff
@@ -16,219 +16,446 @@ import "../../../core/engines/cross-margin//AccountUtil.sol";
...
+        positions[3] = OptionPosition(TokenType.PUT, 18000 * UNIT, 17 * sUNIT);
+
+        Balance[] memory balances = _previewMinCollateral(positions);
+
+        assertEq(balances.length, 1);
+        assertEq(balances[0].collateralId, wethId);
+        assertEq(balances[0].amount, 1 * 1e18);
+    }
+
+    function testPutGreaterThanCalls() public {
```
Comment (dsshap) [inline src/test/engine-integrations/cross-margin/PreviewCollateralReq.t.sol:448]: These are all very deep ITM

**Review by dsshap — CHANGES_REQUESTED** [review]

File: src/test/engine-integrations/cross-margin/PreviewCollateralReq.t.sol (hunk 2 above)
Comment (Fletch153) [reply src/test/engine-integrations/cross-margin/PreviewCollateralReq.t.sol:448]: It was a copy of the test in CMM.t.sol. I've run through a similar unusual scenario and updated the test

File: src/test/engine-integrations/cross-margin/PreviewCollateralReq.t.sol (hunk 1 above)
Comment (Fletch153) [reply src/test/engine-integrations/cross-margin/PreviewCollateralReq.t.sol:236]: I used this formula to calculate how much ETH collateral is required (4 545 454 545 000 000 000 000), as the original formula in CMM.t.sol ((1000_00000 * **10^6**) / 22000) was based on the underlying having a precision of 10^6, but altering it to (1000_00000 * **10^18** / 22000) gives a result of '4 545 454 545 454 545 454 545', which won't work due to precision limits within Grappa.

File: src/test/engine-integrations/cross-margin/PreviewCollateralReq.t.sol (hunk 2 above)
Comment (dsshap) [reply src/test/engine-integrations/cross-margin/PreviewCollateralReq.t.sol:448]: @hashnoteleo  was the intention for this test to be deep ITM? 

https://github.com/antoncoding/grappa/blob/master/src/test/unit-tests/cross-margin/CrossMarginMath.t.sol#L1029

Comment (hashnoteleo) [reply src/test/engine-integrations/cross-margin/PreviewCollateralReq.t.sol:448]: Yes, it was meant to be 1 put otm, 2 calls otm, and 1 put deep deep itm. Wanted to check the crossing of call and put strikes.

**Review by dsshap — APPROVED** [review]

**Pull request merged**